Numpy.py

```diff
@@ -0,0 +1,97 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+# 1
+arr_0_to_20 = np.array(range(21))
+arr_0_to_20_after_scale = np.concatenate((arr_0_to_20[:8], (arr_0_to_20[9:16] * -1), arr_0_to_20[16:]))
+print("tar1: \n", arr_0_to_20_after_scale)
+
+# 2
+arr_div_to_10_part = np.linspace(5, 50, 10)
+print("tar2: \n", arr_div_to_10_part)
+
+# 3
+matrix = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+print("tar3: \n", matrix.shape)
+
+# 4
+matrix_10x10 = np.zeros((10, 10))
+matrix_10x10[0] = np.ones(10)
+matrix_10x10[9] = np.ones(10)
+matrix_10x10[:, 0] = np.ones(10)
+matrix_10x10[:, 9] = np.ones(10)
+print("tar4: \n", matrix_10x10)
+
+# 5
+matrix_to_add_vector = np.zeros((3, 3))
+vec_for_added = np.ones((5))
+matrix_after_added = np.zeros((matrix_to_add_vector.shape[0], matrix_to_add_vector.shape[1] + vec_for_added.shape[0]),int)
+matrix_after_added[:, :matrix_to_add_vector.shape[1]] = matrix_to_add_vector
+matrix_after_added[:, matrix_to_add_vector.shape[1]:] = vec_for_added
```

Comment:
I think you misunderstood the assignment, The point was to add the values of the vector to each of the rows. Either way there is no need to create a new matrix.